src/chatbox/ChatBox.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+  <div id="ChatBox">
+    <chat-box-header />
+    <chat-box-message-list v-if="isJoined" :channelId="channelId" />
+    <chat-box-input :channelId="channelId" />
+  </div>
+</template>
+
+<script>
+import ChatBoxHeader from "@/chatbox/main/ChatBoxHeader.vue";
+import ChatBoxInput from "@/chatbox/main/ChatBoxInput.vue";
+import ChatBoxMessageList from "@/chatbox/main/ChatBoxMessageList.vue";
+
+import { ClientInstance } from "@/chatbox/sdkInstance";
+
+import { EkoChannelType, ChannelRepository } from "eko-sdk";
+const channelRepo = new ChannelRepository();
+
+export default {
+  name: "ChatBox",
+  components: {
+    ChatBoxHeader,
+    ChatBoxInput,
+    ChatBoxMessageList,
+  },
+  props: ["api_key", "userId", "channelId"],
+  data: () => ({
+    current_key: "",
+    isJoined: false,
+  }),
+  watch: {
+    current_key: function (val) {
+      ClientInstance.init(val);
+      ClientInstance.registerUserSession(this.userId);
+      this.joinUserToChannel(this.channelId, EkoChannelType.Standard);
+    },
+  },
+  methods: {
+    joinUserToChannel(channelId, type) {
+      const liveChannel = channelRepo.joinChannel({
+        channelId,
+        type,
+      });
+      if (Object.keys(liveChannel).length > 0) {
+        liveChannel.once("dataUpdated", (data) => {
+          console.log("Channel data update: ", data);
+        });
+        this.isJoined = true;
+      } else {
+        this.isJoined = false;
+      }
```

Comment:
```suggestion
      this.isJoined = false
      
      const liveChannel = channelRepo.joinChannel({ channelId, type });
      liveChannel.once("dataUpdated", (data) => this.isJoined = true)
```

Reply:
From the above comments that remove watch: { as well as methods: { out. So where to put the function in then in 

`  beforeMount() {
    ClientInstance.init(this.api_key);
    ClientInstance.registerUserSession(this.userId);
     this.isJoined = false
     const liveChannel = channelRepo.joinChannel({ channelId, type });
     liveChannel.once("dataUpdated", (data) => this.isJoined = true)
  },`

Reply:
oh shit. let me correct this error in my review.

Reply:
<img width="668" alt="Screen Shot 2564-05-18 at 10 58 05" src="https://user-images.githubusercontent.com/80325355/118588128-062eba80-b7c8-11eb-84d6-d54ea71c566b.png">